lib/galaxy/tool_util/verify/interactor.py

```diff
@@ -174,10 +176,14 @@ def verify_output(self, history_id, jobs, output_data, output_testdef, tool_id,
             raise AssertionError(f"Output {name}: {str(e)}")
 
         primary_datasets = attributes.get('primary_datasets', {})
-        if primary_datasets:
-            job_id = self._dataset_provenance(history_id, hid)["job_id"]
-            outputs = self._get(f"jobs/{job_id}/outputs").json()
-
+        job_id = self._dataset_provenance(history_id, hid)["job_id"]
+        outputs = self._get(f"jobs/{job_id}/outputs").json()
+        found_outputs = 0
```

Comment:
I'd rename this to found_datasets, to discriminate a dataset from a tool output.
```suggestion
        found_datasets = 0
```